src/python/pants/option/global_options.py

```diff
@@ -251,10 +288,26 @@ def register_bootstrap_options(cls, register):
     register('--build-file-imports', choices=['allow', 'warn', 'error'], default='warn',
       help='Whether to allow import statements in BUILD files')
 
-    register('--remote-store-server',
-             help='host:port of grpc server to use as remote execution file store')
-    register('--remote-execution-server',
-             help='host:port of grpc server to use as remote execution scheduler')
+    remote_store_server = '--remote-store-server'
+    remote_execution_server = '--remote-execution-server'
+    register(remote_store_server, advanced=True,
+             help='host:port of grpc server to use as remote execution file store.')
+    register('--remote-store-thread-count', type=int, advanced=True,
+             default=DEFAULT_EXECUTION_OPTIONS.remote_store_thread_count,
+             help='Thread count to use for the pool that interacts with the remote file store.')
+    register(remote_execution_server, advanced=True,
+             help='host:port of grpc server to use as remote execution scheduler.')
+    register('--remote-store-chunk-size', type=int, advanced=True,
+             default=DEFAULT_EXECUTION_OPTIONS.remote_store_chunk_size,
+             help='Size in bytes of chunks transferred to/from the remote file store.')
+    register('--remote-store-chunk-upload-timeout', type=int, advanced=True,
```

Comment:
`-seconds`?

Reply:
+1, it's a good habit to put the units in the name. Although possibly the resolution should be milliseconds. Seconds seems too coarse.

Reply:
> Seconds seems too coarse.

The python convention would be to go to floating point seconds rather than milliseconds. I haven't done that here, but we could in the future if need be.